### PR TITLE
Add tagliatelle linter

### PR DESCRIPTION
**Pull request by ldez**
tagliatelle is a linter that handles struct tags.

for now, it only handle the string casing.

https://github.com/ldez/tagliatelle

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -128,11 +128,6 @@ issues:
       linters:
         - gomnd
 
-    # https://github.com/go-critic/go-critic/issues/926
-    - linters:
-        - gocritic
-      text: "unnecessaryDefer:"
-
```
Comment (bombsimon) [inline .golangci.yml:135]: nit: Would this fit better in a separate PR?

**Review by bombsimon — APPROVED** [review]
Nice linter, lgtm! Would be nice with examples or links to each styling in the upstream repo but that's not really a part of this PR and more a nice to have. Might not be obvious to everyone what the difference between f.ex. `camel` and `goCamel` is.

File: .golangci.yml (hunk 1 above)
Comment (ldez) [reply .golangci.yml:135]: I used that to purge the cache :wink:

**Comment by ldez** [comment]
I improved the readme to add illustrations of the different string casing.

https://github.com/ldez/tagliatelle

**Pull request merged**